### PR TITLE
(icloud) Add 3010 to valid exit codes

### PR DESCRIPTION
**Pull request by brogers5**
## Description
Fairly straightforward change: adds exit code 3010 (`ERROR_SUCCESS_REBOOT_REQUIRED`) to the install script's list of valid exit codes.

## Motivation and Context
I was working on updating a package that has a runtime dependency on a DLL distributed with the non-Microsoft Store version of iCloud (i.e the same installer consumed by the `icloud` package). In my environment, `iCloud64.msi` returned this while testing an upgrade to the latest version available.

Interestingly, while this does not fail the package install itself, the exclusion does cause `Start-ChocolateyProcessAsAdmin` to write a misleading error.

Example log file:
[choco iCloud.log](https://github.com/mkevenaar/chocolatey-packages/files/8934646/choco.iCloud.log)

#### Dev
* OS Build: Windows 11 Pro v10.0.22000.0 (64-bit)
* PowerShell version: 7.2.4
* Chocolatey version: 1.1.0

#### Test
* OS Build: WIndows 10 Pro v10.0.17763.0 (32-bit)
* PowerShell version: 5.1.17763.1490
* Chocolatey version: 1.1.0

**NOTE**: This package is incompatible with the official Chocolatey Testing Environment, since `iCloud64.msi` prevents installation on a server OS.

### Steps
1. Confirm iCloud is NOT currently installed. If currently installed, uninstall it first.
2. Bump Nuspec's version using package fix notation (included in this changeset).
3. Create test package version using `choco pack`.
4. Install an older version of iCloud (e.g. `choco install icloud --version=7.18.0.22`).
5. Reboot to ensure iCloud installation is in a known-good state.
6. Upgrade to test package version (e.g. `choco upgrade icloud --source=.`).
7. Inspect installation output and confirm the following:
    * Package installation succeeds
    * Returned exit code is 3010
    * No errors are written relating to exit code

## Screenshot (if appropriate, usually isn't needed):
N/A

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:
- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/mkevenaar/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [ ] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

### CODE REVIEW

File: automatic/icloud/icloud.nuspec [hunk 1]
```diff
@@ -26,7 +26,7 @@ This is a nuspec. It mostly adheres to https://docs.nuget.org/create/Nuspec-Refe
     <!-- version should MATCH as closely as possible with the underlying software -->
     <!-- Is the version a prerelease of a version? https://docs.nuget.org/create/versioning#creating-prerelease-packages -->
     <!-- Note that unstable versions like 0.0.1 can be considered a released version, but it's possible that one can release a 0.0.1-beta before you release a 0.0.1 version. If the version number is final, that is considered a released version and not a prerelease. -->
-    <version>7.21.0.23</version>
+    <version>7.21.0.2300</version>
```
Comment (mkevenaar) [inline automatic/icloud/icloud.nuspec:29]: Could you please revert this change? Bumping the version number is done using AU

**Review by mkevenaar — CHANGES_REQUESTED** [review]
Hi,

Thank you for adding this, however, could you please revert the version number change?

File: automatic/icloud/icloud.nuspec (hunk 1 above)
Comment (brogers5) [reply automatic/icloud/icloud.nuspec]: Addressed in 167e20c02fb7362f6f75c7d7350814a1187bbf18.

**Review by mkevenaar — APPROVED** [review]
LGTM!

**Comment by brogers5** [comment]
@mkevenaar, I saw [your comment on #135](https://github.com/mkevenaar/chocolatey-packages/pull/135#issuecomment-1160524253) stating this change will propagate with the next package release.

Apple has been funneling Windows 10+ users towards the Microsoft Store version, and only maintains this version for compatibility with legacy Windows systems. Seeing as the last release was about 2 years ago, and that Windows 8.1's EOL date is approaching, I think it's unlikely we'll be seeing another release from Apple.

While this is a minor change that, in my opinion, wouldn't warrant a package fix version, you may want to consider creating one to ensure this doesn't remain in limbo.
